runtime/src/snapshot_utils.rs

```diff
@@ -1139,54 +1140,63 @@ pub(crate) fn parse_incremental_snapshot_archive_filename(
     })
 }
 
-/// Get a list of the full snapshot archives in a directory
-pub fn get_full_snapshot_archives<P>(snapshot_archives_dir: P) -> Vec<FullSnapshotArchiveInfo>
-where
-    P: AsRef<Path>,
-{
-    match fs::read_dir(&snapshot_archives_dir) {
-        Err(err) => {
+/// Walk down the snapshot archive directory recursively to collect snapshot archive file info
+fn get_snapshot_archives<T>(dir: &Path, cb: fn(PathBuf) -> Result<T>, output: &mut Vec<T>) {
```

Comment:
Could this fn be simplified? Getting all the snapshot archives just needs to check an additional "remote" directory in addition to the normal one. And we could go back to returning the results instead of requiring the `&mut` parameter.

Reply:
yeah. only need to look into remote sub directory. Commented by @sakridge as well.